dynatrace/dynatrace.go

```diff
@@ -99,67 +101,49 @@ func (e *Exporter) Export(ctx context.Context, cs export.CheckpointSet) error {
+		dims := []dimensions.Dimension{}
 		for itr.Next() {
 			label := itr.Label()
-			dim := mint.NewDimension(string(label.Key), label.Value.Emit())
-			dimensions = append(dimensions, dim)
+			dims = append(dims, dimensions.NewDimension(string(label.Key), label.Value.Emit()))
+		}
+
+		valOpt, err := getValueOption(r)
+
+		if err != nil {
+			e.logger.Warn(fmt.Sprintf("failed to normalize metric: %s - %s", r.Descriptor().Name(), err.Error()))
+			return nil
 		}
 
-		descriptor := mint.SerializeDescriptor(r.Descriptor().Name(), e.opts.Prefix, dimensions, e.opts.Tags)
-		if descriptor == "" {
-			e.logger.Warn(fmt.Sprintf("failed to normalize metric name: %s", r.Descriptor().Name()))
+		metric, err := dtMetric.NewMetric(
+			r.Descriptor().Name(),
+			dtMetric.WithPrefix(e.opts.Prefix),
+			dtMetric.WithDimensions(
+				dimensions.MergeLists(
+					dimensions.NewNormalizedDimensionList(e.opts.DefaultDimensions...),
+					dimensions.NewNormalizedDimensionList(dims...),
+				),
+			),
+			valOpt,
+		)
+
+		if err != nil {
+			e.logger.Warn(fmt.Sprintf("failed to normalize metric: %s - %s", r.Descriptor().Name(), err.Error()))
 			return nil
 		}
 
-		switch agg := agg.(type) {
-		case aggregation.MinMaxSumCount:
-			min, err := agg.Min()
-			if err != nil {
-				return fmt.Errorf("error getting min for %s: %w", descriptor, err)
-			}
-
-			max, err := agg.Max()
-			if err != nil {
-				return fmt.Errorf("error getting max for %s: %w", descriptor, err)
-			}
-
-			sum, err := agg.Sum()
-			if err != nil {
-				return fmt.Errorf("error getting sum for %s: %w", descriptor, err)
-			}
-
-			count, err := agg.Count()
-			if err != nil {
-				return fmt.Errorf("error getting count for %s: %w", descriptor, err)
-			}
-
-			switch r.Descriptor().NumberKind() {
-			case metric.Float64NumberKind:
-				lines = append(lines, mint.SerializeRecord(descriptor, mint.SerializeDoubleSummaryValue(min.AsFloat64(), max.AsFloat64(), sum.AsFloat64(), count)))
-			case metric.Int64NumberKind:
-				lines = append(lines, mint.SerializeRecord(descriptor, mint.SerializeIntSummaryValue(min.AsInt64(), max.AsInt64(), sum.AsInt64(), count)))
-			}
-
-		case aggregation.Sum:
-			val, err := agg.Sum()
-			if err != nil {
-				return fmt.Errorf("error getting LastValue for %s: %w", descriptor, err)
-			}
-
-			switch r.Descriptor().NumberKind() {
-			case metric.Float64NumberKind:
-				lines = append(lines, mint.SerializeRecord(descriptor, mint.SerializeDoubleCountValue(val.AsFloat64())))
-			case metric.Int64NumberKind:
-				lines = append(lines, mint.SerializeRecord(descriptor, mint.SerializeIntCountValue(val.AsInt64())))
-			}
+		line, err := metric.Serialize()
+
+		if err != nil {
+			e.logger.Warn(fmt.Sprintf("failed to serialize metric: %s - %s", r.Descriptor().Name(), err.Error()))
+			return nil
 		}
 
+		lines = append(lines, line)
+
 		return nil
 	})
+
 	if err != nil {
 		return fmt.Errorf("error generating metric lines: %s", err.Error())
 	}
```

Comment:
Do we want to partition the strings before exporting? Just in case there are more than 1000 per request.